internal/host/plugin/repository_host_set.go

```diff
@@ -338,3 +339,117 @@ func toPluginSet(ctx context.Context, in *HostSet) (*pb.HostSet, error) {
 	}
 	return hs, nil
 }
+
+// Endpoints provides all the endpoints available for a given set id.
+// An error is returned if the set, related catalog, or related plugin are
+// unable to be retrieved.  If a host does not contain an addressible endpoint
+// it is not included in the resulting slice of endpoints.
+func (r *Repository) Endpoints(ctx context.Context, setId string) ([]*host.Endpoint, error) {
```

Comment:
Could this take in a `[]string` of sets so that we could potentially reduce the number of calls to the DB? ListHosts below already takes in multiple sets.

Reply:
Changed to allow a slice of set ids to be passed to it.  I didn't have time to add many tests to exercise that functionality.